mitiq/zne/tests/test_zne.py

```diff
@@ -560,3 +563,51 @@ def qs_noisy_simulation(
 
     mitigated = execute_with_zne(circuit, qs_noisy_simulation)
     assert abs(mitigated) < 1000
+
+
+@pytest.mark.parametrize(
+    "noise_scaling_method",
+    [fold_gates_at_random, insert_id_layers, fold_global, fold_all],
+)
+@pytest.mark.parametrize(
+    "extrapolation_factory", [RichardsonFactory, LinearFactory]
+)
+def test_two_stage_zne(noise_scaling_method, extrapolation_factory):
```

Comment:
I think for now the combined test is good enough. As we iterate in future PRs and the functions get more complicated, it might be better to separate them then.